tests/unit/test_tes_models.py

```diff
@@ -0,0 +1,206 @@
+    executor = TESExecutor(
+        image="python:3.8-slim",
+        command=["python", "script.py"],
+        workdir="/app",
+        stdout="/logs/stdout.log",
+        stderr="/logs/stderr.log",
+        stdin="/input/input.txt",
+        env={"ENV_VAR": "value"},
+    )
+    assert executor.image == "python:3.8-slim"
+    assert executor.command == ["python", "script.py"]
+    assert executor.workdir == "/app"
+    assert executor.stdout == "/logs/stdout.log"
+    assert executor.stderr == "/logs/stderr.log"
+    assert executor.stdin == "/input/input.txt"
+    assert executor.env == {"ENV_VAR": "value"}
+
+
+def test_tes_executor_with_relative_stdin():
+    """Test TESExecutor model with a relative stdin path (should raise ValidationError)."""
+    with pytest.raises(ValidationError) as exc_info:
+        TESExecutor(
+            image="python:3.8-slim",
+            command=["python", "script.py"],
+            stdin="input.txt",
+        )
+    assert "The 'stdin' attribute must contain an absolute path." in str(exc_info.value)
+
+
+def test_tes_data_valid():
+    """Test TESData model with valid data."""
+    tes_data = TESData(
+        id="task-123",
+        name="Test Task",
+        description="An example TES task.",
+        creation_time=datetime.utcnow().replace(tzinfo=timezone.utc),
+        state=TESState.QUEUED,
+        inputs=[
+            TESInput(
+                name="Test Input",
+                description="An example input file.",
+                url="https://raw.githubusercontent.com/elixir-cloud-aai/CrateGen/refs/heads/main/README.md",
+                path="/data/input/README.md",
+                type=TESFileType.FILE,
+            )
+        ],
+        outputs=[
+            TESOutput(
+                name="Test Output",
+                description="An example output file.",
+                url="https://raw.githubusercontent.com/elixir-cloud-aai/CrateGen/refs/heads/main/LICENSE",
+                path="/data/output/LICENSE",
+                type=TESFileType.FILE,
+            )
+        ],
+        executors=[
+            TESExecutor(
+                image="python:3.8-slim",
+                command=["python", "script.py"],
+            )
+        ],
+        resources=TESResources(
+            cpu_cores=EXPECTED_CPU_CORES,
+            ram_gb=EXPECTED_RAM_GB,
+            disk_gb=EXPECTED_DISK_GB,
+            preemptible=EXPECTED_PREEMPTIBLE,
+        ),
+        volumes=["/data"],
+        tags={"project": "CrateGen"},
+    )
+    assert tes_data.id == "task-123"
+    assert tes_data.inputs[0].url == "https://raw.githubusercontent.com/elixir-cloud-aai/CrateGen/refs/heads/main/README.md"
+    assert tes_data.outputs[0].path == "/data/output/LICENSE"
+    assert tes_data.executors[0].image == "python:3.8-slim"
+    assert tes_data.resources.cpu_cores == EXPECTED_CPU_CORES
+    assert tes_data.volumes == ["/data"]
+    assert tes_data.tags == {"project": "CrateGen"}
```

Comment:
I agree that the `test_tes_data_valid` function is quite extensive. I will refactor it by creating separate, focused tests for each subcomponent (`inputs`, `outputs`, `executors`, `resources`). This will improve test readability and maintainability.